src/mixer/basetrackplayer.cpp

```diff
@@ -134,8 +136,37 @@ BaseTrackPlayerImpl::BaseTrackPlayerImpl(QObject* pParent,
             SLOT(slotWaveformZoomSetDefault(double)));
 
     m_pPreGain = std::make_unique<ControlProxy>(group, "pregain", this);
-    // BPM of the current song
 
+    m_pShiftTimesEarlier = std::make_unique<ControlPushButton>(
+            ConfigKey(group, "shift_times_earlier"));
```

Comment:
What is "times" that is shifted? Aren't the cue points shifted under the notation? 
"shifft_all_cues_earlier"

Reply:
I kept the name generic with the intention of also shifting beatgrids when we import those. Right now we are only importing beatgrids from Rekordbox (see discussion above).

Reply:
We already have controls to shift the beatgrids. 

I can't imagine how a generic control can work here. Do you? So Yagni.

Instead we could use the imported beatgrids as input for our (new) beat detector. 

Even in Mixxx we have the use case that once a new beat detector is in place the question comes up if we should move quantized cues to the new beats.

We even have the use case that the cued imported via a perfectly offset adjusted algorithm will be off the beats, because the beatgrid was slightly off in the source tool.

Reply:
> Even in Mixxx we have the use case that once a new beat detector is in place the question comes up if we should move quantized cues to the new beats.

I had not thought of any other use cases for this, but this would make sense. However, changing users' data without explicit permission is unacceptable.